### PR TITLE
Refactor documentation

### PR DESCRIPTION
**Pull request by mrd0ll4r**

### CODE REVIEW

File: middleware/middleware.go [hunk 1]
```diff
@@ -12,12 +12,14 @@ import (
 	"github.com/chihaya/chihaya/storage"
 )
 
+// Config holds the configuration of a tracker's Logic.
```
Comment (jzelinskie) [inline middleware/middleware.go]: Config holds the configuration common across all middleware.

File: middleware/middleware.go [hunk 2]
```diff
@@ -12,12 +12,14 @@ import (
...
 type Config struct {
 	AnnounceInterval time.Duration `yaml:"announce_interval"`
 }
 
 var _ frontend.TrackerLogic = &Logic{}
 
+// NewLogic creates a new Logic from the given parameters.
 func NewLogic(cfg Config, peerStore storage.PeerStore, preHooks, postHooks []Hook) *Logic {
```
Comment (jzelinskie) [inline middleware/middleware.go]: NewLogic creates a new instance of a TrackerLogic that executes the provided middleware hooks.

**Comment by jzelinskie** [comment]
If you add a line to your commit message at the bottom that says `Fixes #46`, it'll auto-close it when this gets merged into master.

**Comment by mrd0ll4r** [comment]
Can you take another look please?

**Comment by jzelinskie** [comment]
LGTM!

**Pull request merged**